lib/dtutils/extensions.lua

```diff
@@ -0,0 +1,185 @@
+--[[
+
+  NAME
+    dtutils/extensions.lua - a library of extenstions to the lua libraries
+
+  SYNOPSIS
+    require "lib/dtutils.extensions"
+
+  DESCRIPTION
+    This library contains extensions to the lua libraries.
+
+]]
+
+local dtutils_extensions = {}
+dtutils_extensions.libdoc = {
+  Sections = {"Name", "Synopsis", "Description", "License"},
+  Name = "dtutils.extensions - a library of extenstions to the lua libraries",
+  Synopsis = [[require "lib/dtutils.extensions"]],
+  Description = [[This library contains extensions to the lua libraries.]],
+  License = [[This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.]],
+  functions = {}
+}
+
+--[[
+  
+  NAME
+    string.strip_accents - strip accents from characters
+
+  SYNOPSIS
+    require "lib/dtutils.extensions"
+
+    result = string.strip_accents(str)
+      str - string - the string with characters that need accents removed
+
+  DESCRIPTION
+    strip_accents removes accents from accented characters returning the 
+    unaccented character.
+
+  RETURN VALUE
+    result - string - the string containing unaccented characters
+  
+]]
+
+dtutils_extensions.libdoc.functions[#dtutils_extensions.libdoc.functions + 1] = {
+  Sections = {"Name", "Synopsis", "Description", "Return_Value"},
+  Name = [[string.strip_accents - strip accents from characters]],
+  Synopsis = [[require "lib/dtutils.extensions"
+
+    result = string.strip_accents(str)
+      str - string - the string with characters that need accents removed]],
+  Description = [[strip_accents removes accents from accented characters returning the 
+    unaccented character.]],
+  Return_Value = [[result - string - the string containing unaccented characters]],
+}
+
+-- Strip accents from a string
+-- Copied from https://forums.coronalabs.com/topic/43048-remove-special-characters-from-string/
+function string.strip_accents( str )
```

Comment:
I just copied these out of the scripts that I found them in because I thought they could/would be useful.  I didn't really apply a lot of thought about how they fit into the grand scheme of things.  That's why I like having others look over what I did and saying "Did you really think this out?" because sometimes I don't ;).  I kind of struggled with what to do with these routines anyway and dtutils.extensions was the best I could come up with.  dtutils.string makes more sense and gives us a home for future string manipulation routines.